src/main/scala/com/lohika/vteraz/persistence/repository/SlickH2UserRepository.scala

```diff
@@ -0,0 +1,28 @@
+package com.lohika.vteraz.persistence.repository
+
+import com.lohika.vteraz.model.UserModel
+import com.lohika.vteraz.persistence.entity.UserTable
+import com.lohika.vteraz.service.RetryService
+import slick.jdbc.H2Profile.api._
+import slick.lifted.TableQuery
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class SlickH2UserRepository(db: Database) extends UserRepository[Future] {
+  val usersTable = TableQuery[UserTable]
+
+  override def registerUser(user: UserModel): Future[Long] = {
+    val userId = (usersTable returning usersTable.map(_.id)) += user
+    RetryService.retryAsync[Long](() => db.run(userId), id => !id.isNaN, List(0.seconds, 1.seconds, 5.seconds))
```

Comment:
you should use slightly modified version of retry async which retries some DB SQL exceptions.